front/packages/ui/src/browse/header.tsx

```diff
@@ -60,79 +50,110 @@ const SortTrigger = forwardRef<View, PressableProps & { sortKey: string }>(funct
 	);
 });
 
+const MediaTypeTrigger = forwardRef<View, PressableProps & { mediaType: MediaType }>(
+	function MediaTypeTrigger({ mediaType, ...props }, ref) {
+		const { css } = useYoshiki();
+		const { t } = useTranslation();
+		const labelKey =
+			mediaType !== MediaTypeAll ? `browse.mediatypekey.${mediaType.key}` : "browse.mediatypelabel";
+		const icon = mediaType !== MediaTypeAll ? mediaType?.icon ?? FilterList : FilterList;
+		return (
+			<PressableFeedback
+				ref={ref}
+				{...css({ flexDirection: "row", alignItems: "center" }, props as any)}
+				{...tooltip(t("browse.mediatype-tt"))}
+			>
+				<Icon icon={icon} {...css({ paddingX: ts(0.5) })} />
+				<P>{t(labelKey as any)}</P>
```

Comment:
I don't think you need the `as any` anymore. Typescript should be able to infer labelKey as a valid translation key (if not please dismiss this comment)

Reply:
Unfortunately it still seems required. I'm not sure if it can infer the type because I'm using string interpolation. I'm not sure if there is a way to make typescript play ball. 🤔

Reply:
I thought `mediaType.key` was known to be a string literal union and typescript would be smart enough. Too bad, let's keep the `as any`